src/utils.ts

```diff
@@ -25,7 +27,11 @@ const VALID_ELEMENTS = [
 ];
 
 class ReactElementTypeError extends Error {
-  constructor(received, matcherFn, context) {
+  constructor(
+    received: ReactTestInstance | null,
+    matcherFn: Function,
```

Comment:
Maybe jest.CustomMatcher would do? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bf0eabf1a39b3eefe344695b27694677cc177a70/types/jest/index.d.ts#L567

Reply:
Changed it